packages/web3-eth-accounts/src/account.ts

```diff
@@ -1,22 +1,173 @@
+};
+
+/**
+ * Signs arbitrary data. The value passed as the data parameter will be UTF-8 HEX decoded and wrapped as follows: "\x19Ethereum Signed Message:\n" + message.length + message
+ */
+export const sign = (data: string, privateKey: string): signResult => {
+	// 64 hex characters + hex-prefix
+	if (privateKey.length !== 66) {
+		throw new PrivateKeyLengthError();
+	}
+
+	const hash = hashMessage(data);
+
+	const signObj = ecdsaSign(
+		Buffer.from(hash.substring(2), 'hex'),
+		Buffer.from(privateKey.substring(2), 'hex'),
+	);
+
+	const r = Buffer.from(signObj.signature.slice(0, 32));
+	const s = Buffer.from(signObj.signature.slice(32, 64));
+	const v = signObj.recid + 27;
+
+	return {
+		message: data,
+		messageHash: hash,
+		v: `0x${v.toString(16)}`,
+		r: `0x${r.toString('hex')}`,
+		s: `0x${s.toString('hex')}`,
+		signature: `0x${Buffer.from(signObj.signature).toString('hex')}${v.toString(16)}`,
+	};
+};
+
+/**
+ *  Signs an Ethereum transaction with a given private key.
+ */
+export const signTransaction = (
+	transaction: TxData | AccessListEIP2930TxData | FeeMarketEIP1559TxData,
+	privateKey: string,
+): signTransactionResult => {
+	//	TO DO : Send calls to web3.transaction package for :
+	//		Transaction Validation checks
+
+	const tx = TransactionFactory.fromTxData(transaction);
+	const signedTx = tx.sign(Buffer.from(privateKey.substring(2), 'hex'));
+	if (signedTx.v === undefined || signedTx.r === undefined || signedTx.s === undefined)
+		throw new SignerError('Signer Error ');
+
+	const validationErrors = signedTx.validate(true);
+
+	if (validationErrors.length > 0) {
+		let errorString = 'Signer Error ';
+		for (const validationError of validationErrors) {
+			errorString += `${errorString} ${validationError}.`;
+		}
+		throw new SignerError(errorString);
+	}
+
+	const rlpEncoded = signedTx.serialize().toString('hex');
+	const rawTx = `0x${rlpEncoded}`;
+	const txHash = keccak256(Buffer.from(rawTx, 'hex'));
+
+	return {
+		messageHash: `0x${Buffer.from(signedTx.getMessageToSign(true)).toString('hex')}`,
+		v: `0x${signedTx.v.toString('hex')}`,
+		r: `0x${signedTx.r.toString('hex')}`,
+		s: `0x${signedTx.s.toString('hex')}`,
+		rawTransaction: rawTx,
+		transactionHash: `0x${Buffer.from(txHash).toString('hex')}`,
+	};
+};
+
+/**
+ * Recovers the Ethereum address which was used to sign the given RLP encoded transaction.
+ */
+export const recoverTransaction = (rawTransaction: HexString): Address => {
+	if (rawTransaction === undefined) throw new UndefinedRAWTransactionError();
+
+	const tx = TransactionFactory.fromSerializedData(Buffer.from(rawTransaction.slice(2), 'hex'));
+
+	return toChecksumAddress(tx.getSenderAddress().toString());
+};
+
+/**
+ * Recovers the Ethereum address which was used to sign the given data
+ */
+export const recover = (
+	data: string | signatureObject,
+	signature?: string,
+	hashed?: boolean,
+): Address => {
+	if (typeof data === 'object') {
```

Comment:
data should be validated, otherwise lgtm

Reply:
`hashMessage` have some data validation and it throws error